app/Export/GetImages.hs

```diff
@@ -28,101 +28,94 @@ getActiveGraphImage req = do
--- [("STA355H1","LEC-0101","F"),("CSC108H1","LEC-0102","F")], [])
-parseCourseCookie :: String -> String -> ([CourseInfo], [CourseInfo])
+        coursecookie = maybe "" cookieValue $ M.lookup "selected-selectedLecs" cookies
+        (selectedLecs, selectedTuts) = parseCourseCookie coursecookie termSession
+    (lecTimes, tutTimes) <- getTimes (selectedLecs, selectedTuts)
+    let schedule = getScheduleByTime selectedLecs selectedTuts lecTimes tutTimes
+    print schedule
+    generateTimetableImg schedule termSession
+
+-- | Parses cookie string and returns two lists of information about courses
+-- in the format of (code, section, session).
+-- One for lecture, the other for tutorial.
+parseCourseCookie :: String -> String -> ([(String, String, String)], [(String, String, String)])
+parseCourseCookie "" _ = ([], [])
 parseCourseCookie s termSession =
   let lecAndTut = map (splitOn "-") $ splitOn "_" s
-      (lecture, tutorial) = partition isLec lecAndTut
+      (selectedLecs, selectedTuts) = partition isLec lecAndTut
       -- get lecture and tutorial in this session
-      [lectureOfsession, tutorialOfsession] = map (filter (\x -> or ([(x !! 2 !! 0) == (head termSession), (x !! 2 !! 0) == 'Y']))) [lecture, tutorial]
-      lecture' = map (convertLec . list2tuple) lectureOfsession
-      tutorial' = map (convertTut . list2tuple) tutorialOfsession
-  in (lecture', tutorial')
+      [lectureOfSession, tutorialOfSession] = map (filter (\x -> or ([(x !! 2 !! 0) == (head termSession), (x !! 2 !! 0) == 'Y']))) [selectedLecs, selectedTuts]
+      selectedLecs' = map (convertLec . list2tuple) lectureOfSession
+      selectedTuts' = map (convertTut . list2tuple) tutorialOfSession
+  in (selectedLecs', selectedTuts')
   where isLec x = (x !! 1 !! 0) == 'L'
 
 list2tuple :: [String] -> (String, String, String)
 list2tuple [a, b, c] = (a, b, c)
 list2tuple _ = undefined
 
--- ("CSC148H1","L5101","S") -> CourseInfo {code = "CSC148H1", section = "LEC-0501", session = "S", time = []} convert format to match database
-convertLec :: (String, String, String) -> CourseInfo
-convertLec (lecCode, lecSection, lecSession) = CourseInfo {
-  code = lecCode,
-  section = take 1 lecSection ++ "EC-" ++ drop 1 lecSection,
-  session = lecSession,
-  time = []
-  }
-
--- ("CSC148H1","T0501","S") -> CourseInfo {code = "CSC148H1", section = "TUT-0501", session = "S", time = []} convert format to match database
-convertTut :: (String, String, String) -> CourseInfo
-convertTut (tutCode, tutSection, tutSession) = CourseInfo {
-  code = tutCode,
-  section = take 1 tutSection ++ "UT-" ++ drop 1 tutSection,
-  session = tutSession,
-  time = []
-  }
-
-
--- | Get data from database for selected courses
-getTimes :: ([CourseInfo], [CourseInfo]) -> IO ([CourseInfo], [CourseInfo])
-getTimes (selectedLectures, selectedTutorials) = runSqlite databasePath $ do
+convertLec :: (String, String, String) -> (String, String, String)
+convertLec (lecCode, lecSection, lecSession) =
```

Comment:
You don't need these helpers; do the conversion directly in `getLectureTime` and `getTutorialTime`.